### PR TITLE
PEK 850: Adding part of the higher priority tests

### PR DESCRIPTION
**Pull request by spmcmillan**
Adding tests to tdx set.  Still need to add to checkbox tests in a separate PR with the rest of the tests specified in PEK850 in next pulse.

Fixes PEK-850

### CODE REVIEW

File: tests/tests/test_guest_fail.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    tdx_NOEPT test case (See https://github.com/intel/tdx/wiki/Tests)
+    """
+
+    # Get initial dmesg contents for comparison later
+    cs = subprocess.run(['sudo', 'dmesg'], check=True, capture_output=True)
+    assert cs.returncode == 0, 'Failed getting dmesg'
+    dmesg_start = str(cs.stdout)
+
+    # Setup ept=0 in driver
+    cs = subprocess.run(['sudo', 'rmmod', 'kvm_intel'], check=True)
```
Comment (hector-cao) [inline tests/tests/test_guest_fail.py:41]: the rmmod will fail if there is any VM running, `rmmod: ERROR: Module kvm_intel is in use` ; i dont know how we can  handle this situation

**Review by hector-cao — CHANGES_REQUESTED** [review]

**Review by spmcmillan — COMMENTED** [review]
LGTM, I didn't approve yet because of the init command pending item, but will if that is ok

File: tests/tests/setup-env-tox.sh [hunk 2]
```diff
@@ -38,7 +38,12 @@ cleanup() {
 install_deps() {
   sudo apt install -y python3-parameterized \
          sshpass \
+         cpuid \
          python3-cpuinfo  &> /dev/null
+  # Make sure kvm_intel is installed properly
+  # (one of the tests installed it with NOEPT)
+  sudo rmmod kvm_intel
```
[the remark this answers is not in the document]
Comment (spmcmillan) [reply tests/tests/setup-env-tox.sh]: It says in the comment, but we load it differently at times on the different tests.  So, this is to make sure we load it the same way at the beginning of each test.

File: tests/tests/test_guest_cpu_off.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+        m = Qemu.QemuSSH(qm)
+
+        # pin pid to a particular cpu
+        cs = subprocess.run(['sudo', 'taskset', '-pc', '18', f'{pid}'], capture_output=True)
+        assert cs.returncode == 0, 'Failed pinning qemu pid to cpu 18'
+
+        cpu_on_off('/sys/devices/system/cpu/cpu18/online', 0)
+
+        m.check_exec('sudo init 0 &')
```
[the remark this answers is not in the document]
Comment (hector-cao) [reply tests/tests/test_guest_cpu_off.py:76]: ok great, thanks

File: tests/tests/setup-env-tox.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (hector-cao) [reply tests/tests/setup-env-tox.sh]: ideally, each test should reload the kernel module properly if it previously unloaded it, but in practice, if there is any reason you cannot have this certainty, loading it in the tox global setup will not be enough, we have to make sure to do it at the beginning of each test. So i think, for now, i will go for the ideal situation where each test should unload/reload the kernel module properly

[the remark this answers is not in the document]
Comment (spmcmillan) [reply tests/tests/setup-env-tox.sh]: true and agreed

**Pull request merged**